akka-stream/src/main/scala/akka/stream/impl/streamref/SinkRefImpl.scala

```diff
@@ -136,7 +148,8 @@ private[stream] final class SinkRefStageImpl[In] private[akka] (
             s"[$stageActorName] Remote side did not subscribe (materialize) handed out Source reference [${promise.future.value}], " +
               s"within subscription timeout: ${PrettyDuration.format(subscriptionTimeout.timeout)}!")
 
-          throw ex // this will also log the exception, unlike failStage; this should fail rarely, but would be good to have it "loud"
+          getPartnerRef ! StreamRefsProtocol.RemoteStreamFailure(ex.getMessage)
```

Comment:
is partnerRef always known when SubscriptionTimeout?

Reply:
Thanks good catch, this is wrong -- we never will have the ref here; the timeout triggers when we did NOT get the ref after all.